scripts/third_party_size_check_test.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2022 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests for scripts/third_party_size_check.py"""
+
+from __future__ import annotations
+
+import builtins
+import os
+import shutil
+import sys
+
+from core import utils
+from core.tests import test_utils
+from scripts import third_party_size_check
+
+from typing import Any
+
+
+class ThirdPartySizeCheckTests(test_utils.GenericTestBase):
+    """Unit tests for scripts/third_party_size_check.py"""
+
+    def test_get_skip_files_list(self) -> None:
+        skip_files_list = (
+            'random_file.py\n'
+            '# This is a comment\n'
+            'new_file.py')
+        with open('dummy_skip_file_list.txt', 'w', encoding='utf-8') as f:
+            f.write(skip_files_list)
+
+        dummy_file_object = open(
+            'dummy_skip_file_list.txt', 'r', encoding='utf-8')
+
+        swap_open = self.swap_with_checks(
+            utils, 'open_file',
+            lambda *unused_args, **unused_kwargs: dummy_file_object,
+            expected_args=(('.gcloudignore', 'r'),))
+
+        with swap_open:
+            returned_list = third_party_size_check._get_skip_files_list() # pylint: disable=protected-access
+
+        skipped_files = ['random_file.py', 'new_file.py']
+        expected_skipped_files_list = [
+            os.path.join(os.getcwd(), file) for file in skipped_files]
+        print(expected_skipped_files_list)
+        self.assertEqual(returned_list, expected_skipped_files_list)
+
+        dummy_file_object.close()
+        os.remove('dummy_skip_file_list.txt')
+
+    def test_get_skip_files_list_throws_error(self) -> None:
+        err = IOError('XYZ error.')
+        print_swap = self.swap_with_checks(
+            builtins, 'print', lambda _: None,
+            expected_args=((err,),))
+        def mock_open_file(*unused_args: Any) -> None:
+            raise err
+        swap_open = self.swap_with_checks(
+            utils, 'open_file', mock_open_file,
+            expected_args=(('.gcloudignore', 'r'),))
+        swap_sys_exit = self.swap(sys, 'exit', lambda _: None)
```

Comment:
Can you use `swap_with_checks` to make sure `sys.exit` is called with the right argument?

Reply:
Done

Reply:
Done